conans/client/toolchain/cmake.py

```diff
@@ -301,28 +352,37 @@ def write_toolchain_files(self):
 
         set_libcxx, glibcxx = self._get_libcxx()
 
+        parallel = None
+        if self._parallel:
+            if self._generator and "Visual Studio" in self._generator:
+                parallel = "/MP%s" % cpu_count(output=self._conanfile.output)
+
+        cppstd, cppstd_extensions = self._cppstd()
+
+        skip_rpath = True if self._conanfile.settings.get_safe("os") == "Macos" else False
+        architecture = self._get_architecture()
```

Comment:
Is there a reason why these specific items are not evaluated in the constructor? There are a bunch of similar items which are.

Reply:
Yes, this will change. Everything will be evaluated in the constructor, so it can be modified later if necessary. I will do this when exposing these things to the UI.

This PR is focused on flattening the template code, moving code to python and removing ``CMakeDefinitionsBuilder``, which was too coupled to the previous CMake build helper and doing too much.